plugin/src/test/java/org/opensearch/ml/model/MLModelManagerTests.java

```diff
@@ -449,6 +451,23 @@ public void testRegisterMLRemoteModel() throws PrivilegedActionException {
         verify(mlTaskManager).updateMLTask(anyString(), anyMap(), anyLong(), anyBoolean());
     }
 
+    public void testRegisterMLRemoteModel_WhenMemoryCBOpen_ThenFail() throws PrivilegedActionException {
```

Comment:
[nit] should it throw `PrivilegedActionException`?

Reply:
does this test fail, if we put `checkAndAddRunningTask(mlTask, maxRegisterTasksPerNode);` outside of the thread context?

Reply:
guess not :)

Reply:
> guess not :)

Is that a answer for both of these two questions?

Reply:
just an answer to the first. my browser seems to be a rather selective comment-loader. The test fails when you put the `checkAnd...` outside of the try block